server/server.py

```diff
@@ -48,15 +51,35 @@ def handle(self):
 
 def keyGen(path):
 	key = RSA.generate(2048)
-	with open(path +'/python.pem','w') as privateKey:
+	with open(path +'/python.pem','wb') as privateKey:
 		privateKey.write(key.exportKey('PEM'))
-	with open(path+ '/python.pub', 'w') as publicKey:
-		publicKey.write(key.publickey.exportKey('PEM'))
+	with open(path+ '/python.pub', 'wb') as publicKey:
+		publicKey.write(key.publickey().exportKey('PEM'))
+
+def parseArgs():
+	'''Parses args using the argparse lib'''
+	parser = argparse.ArgumentParser(description='Location logging server')
+
+	parser.add_argument('-g', '--generate-keys', metavar='PATH', type=str)
```

Comment:
actually we want the opposite. I had added nargs when I was toying with the idea of listening on multiple ports/IPs. With nargs the option is returned as a list, which seems weird for a single item. changes have been pushed to remove nargs.